# --- ROCK-COPYRIGHT-NOTE-BEGIN ---
# 
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# Please add additional copyright information _after_ the line containing
# the ROCK-COPYRIGHT-NOTE-END tag. Otherwise it might get removed by
# the ./scripts/Create-CopyPatch script. Do not edit this copyright text!
# 
# ROCK Linux: rock-src/package/base/lvm2/linux_prio_tree_h.patch
# ROCK Linux is Copyright (C) 1998 - 2005 Clifford Wolf
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# 
# --- ROCK-COPYRIGHT-NOTE-END ---

diff -dur LVM2.2.00.25/lib/device/dev-io.c LVM2.2.00.25-patch/lib/device/dev-io.c
--- LVM2.2.00.25/lib/device/dev-io.c	2004-09-14 22:23:23.000000000 +0000
+++ LVM2.2.00.25-patch/lib/device/dev-io.c	2004-10-23 21:47:37.092530280 +0000
@@ -31,6 +31,13 @@
 #  define u64 uint64_t		/* Missing without __KERNEL__ */
 #  undef WNOHANG		/* Avoid redefinition */
 #  undef WUNTRACED		/* Avoid redefinition */
+
+/* for linux-2.6.9, from linux/types.h */
+#  include <asm/types.h>
+#  ifndef pgoff_t
+#  define pgoff_t unsigned long
+#  endif
+
 #  include <linux/fs.h>		/* For block ioctl definitions */
 #  define BLKSIZE_SHIFT SECTOR_SHIFT
 #  ifndef BLKGETSIZE64		/* fs.h out-of-date */
